src/com/owncloud/android/lib/resources/files/GetMetadataOperation.java

```diff
@@ -0,0 +1,105 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.lib.resources.files;
+
+import com.owncloud.android.lib.common.OwnCloudClient;
+import com.owncloud.android.lib.common.operations.RemoteOperation;
+import com.owncloud.android.lib.common.operations.RemoteOperationResult;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+
+/**
+ * Remote operation performing the fetch of metadata for a folder
+ */
+
+public class GetMetadataOperation extends RemoteOperation {
+
+    private static final String TAG = GetMetadataOperation.class.getSimpleName();
+    private static final int SYNC_READ_TIMEOUT = 40000;
+    private static final int SYNC_CONNECTION_TIMEOUT = 5000;
+    private static final String METADATA_URL = "/ocs/v2.php/apps/end_to_end_encryption/api/v1/meta-data/";
+
+    // JSON node names
+    private static final String NODE_OCS = "ocs";
+    private static final String NODE_DATA = "data";
+    private static final String NODE_META_DATA = "meta-data";
+
+    private static final String JSON_FORMAT = "?format=json";
+
+    private String fileId;
+
+    /**
+     * Constructor
+     */
+    public GetMetadataOperation(String fileId) {
+        this.fileId = fileId;
+    }
+
+    /**
+     * @param client Client object
+     */
+    @Override
+    protected RemoteOperationResult run(OwnCloudClient client) {
+        GetMethod getMethod = null;
+        RemoteOperationResult result;
+
+        try {
+            // remote request
+            getMethod = new GetMethod(client.getBaseUri() + METADATA_URL + fileId + JSON_FORMAT);
+            getMethod.addRequestHeader(OCS_API_HEADER, OCS_API_HEADER_VALUE);
+
+            int status = client.executeMethod(getMethod, SYNC_READ_TIMEOUT, SYNC_CONNECTION_TIMEOUT);
+
+            if (status == HttpStatus.SC_OK) {
+                String response = getMethod.getResponseBodyAsString();
+
+                // Parse the response
+                JSONObject respJSON = new JSONObject(response);
+                String metadata = (String) respJSON.getJSONObject(NODE_OCS).getJSONObject(NODE_DATA)
+                        .get(NODE_META_DATA);
+
+                result = new RemoteOperationResult(true, getMethod);
+                ArrayList<Object> metadataArray = new ArrayList<>();
+                metadataArray.add(metadata);
+                result.setData(metadataArray);
+            } else {
+                result = new RemoteOperationResult(false, getMethod);
+                client.exhaustResponse(getMethod.getResponseBodyAsStream());
+            }
+        } catch (Exception e) {
+            result = new RemoteOperationResult(e);
+            e.printStackTrace();
```

Comment:
Please use a logger instead :)

Reply:
nevermind, it is just in the next line.

Reply:
👍 still it is not useful to print a stacktrace.